output/cloud/expv2/metrics_client.go

```diff
@@ -5,64 +5,46 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"io"
 	"net/http"
-	"sync"
-	"time"
+	"strings"
 
 	"github.com/klauspost/compress/snappy"
-	"github.com/sirupsen/logrus"
 	"google.golang.org/protobuf/proto"
 
 	"go.k6.io/k6/cloudapi"
-	"go.k6.io/k6/lib/consts"
 	"go.k6.io/k6/output/cloud/expv2/pbcloud"
 )
 
-type httpDoer interface {
-	Do(*http.Request) (*http.Response, error)
-}
-
 // metricsClient is a Protobuf over HTTP client for sending
 // the collected metrics from the Cloud output
 // to the remote service.
 type metricsClient struct {
-	httpClient httpDoer
-	logger     logrus.FieldLogger
-	token      string
-	userAgent  string
-
-	pushBufferPool sync.Pool
-	baseURL        string
+	httpClient *cloudapi.Client
+	baseURL    string
 }
 
 // newMetricsClient creates and initializes a new MetricsClient.
-func newMetricsClient(logger logrus.FieldLogger, host string, token string) (*metricsClient, error) {
-	if host == "" {
-		return nil, errors.New("host is required")
-	}
-	if token == "" {
-		return nil, errors.New("token is required")
+func newMetricsClient(c *cloudapi.Client) (*metricsClient, error) {
+	// Unfortunately, the cloudapi.Client works across different versions
+	// of the API, but it has the v1 harcoded so we need to trim the wrong path
+	// to be able to replace it with the correct one.
+	u := c.BaseURL()
+	if !strings.HasSuffix(u, "/v1") {
+		return nil, errors.New("a /v1 suffix is expected in the Cloud service's BaseURL path")
 	}
 	return &metricsClient{
-		httpClient: &http.Client{Timeout: 5 * time.Second},
-		logger:     logger,
-		baseURL:    host + "/v2/metrics/",
-		token:      token,
-		userAgent:  "k6cloud/v" + consts.Version,
-		pushBufferPool: sync.Pool{
-			New: func() interface{} {
-				return &bytes.Buffer{}
-			},
-		},
+		httpClient: c,
+		baseURL:    strings.TrimSuffix(u, "/v1") + "/v2/metrics/",
 	}, nil
 }
 
-// Push pushes the provided metrics the given test run.
-func (mc *metricsClient) push(ctx context.Context, referenceID string, samples *pbcloud.MetricSet) error {
+// Push pushes the provided metrics the given test run. Inject a timeout context
+// for limiting the request duration.
+func (mc *metricsClient) push(referenceID string, samples *pbcloud.MetricSet) error {
```

Comment:
There's no context argument anymore, so let's remove the comment.

```suggestion
// Push the provided metrics for the given test run ID.
func (mc *metricsClient) push(referenceID string, samples *pbcloud.MetricSet) error {
```

Reply:
Done

Reply:
There were some suggestions for the first sentence there as well, but OK.

Reply:
It should be applied now